### PR TITLE
logger: implement logger option

### PR DESCRIPTION
**Pull request by lopezator**
Add a Logger interface, along with a LoggerFunc and a new WithLogger option.

This would enable:

- Customizing migrator output
- Integrating a 3rd party logging tool

By default, migrator will work as before, just outputting to stdout.

Closes #6

### CODE REVIEW

**Comment by lopezator** [comment]
Would this be enough for you @adamdecaf?

You could use `WithLogger` option along with `LoggerFunc` function, and then do nothing inside to avoid printing.

```go
migrator, err := New(Migrations(migrations...), WithLogger(LoggerFunc(func(string, ...interface{}) {
    // Doing nothing here would disable the output
})))
```

CC/ @dufcrule @glerchundi

**Comment by adamdecaf** [comment]
It also works to reformat logs into JSON, thanks!

**Comment by lopezator** [comment]
That's great! Did you used some 3rd party tool for json formatting?

**Comment by adamdecaf** [comment]
We use the [go-kit/kit logger](https://pkg.go.dev/github.com/go-kit/kit@v0.10.0/log?tab=doc) which has a JSON outputter.

**Comment by lopezator** [comment]
That's great. Thank you for testing it out!

File: README.md [hunk 1]
```diff
@@ -77,8 +77,23 @@ func main() {
 ```
 
 Notes on examples above:
+
 - Migrator creates/manages a table named `migrations` to keep track of the applied versions. However, if want to customize the table name `migrator.TableName("my_migrations")` can be passed to `migrator.New` function as an additional option.
```
Comment (dufcrule) [inline README.md:81]: However, if `you` want to customize the table name...

File: README.md [hunk 2]
```diff
@@ -77,8 +77,23 @@ func main() {
 ```
 
 Notes on examples above:
+
 - Migrator creates/manages a table named `migrations` to keep track of the applied versions. However, if want to customize the table name `migrator.TableName("my_migrations")` can be passed to `migrator.New` function as an additional option. 
 
+### Logging
+
+By default, migrator prints applying/applied migration info to stdout, it that's enough for you, you can skip this section.
```
Comment (dufcrule) [inline README.md:85]: Typo and should be two sentences:  By default, migrator prints applying/applied migration info to stdout. `If` that's enough for you, you can skip this section.

File: README.md [hunk 3]
```diff
@@ -77,8 +77,23 @@ func main() {
...
+
+If you need some special formatting or want to use a 3rd party logging library, this could be done by using `WithLogger` option as follows:
+
+```go
+logger := migrator.WithLogger(migrator.LoggerFunc(func(str string, args ...interface{}) {
+	// Your code here 
+})))
+```
+
+Then you only will need to pass the logger as an option to `migrator.New`.
```
Comment (dufcrule) [inline README.md:95]: Word order improvement: Then you will only need to pass...

**Comment by dufcrule** [comment]
Just some minor changes to the README texts. Apart from that LGTM!

**Comment by lopezator** [comment]
Thank you for the review @dufcrule !!

**Pull request merged**